docs/getting_started.rst

```diff
@@ -31,105 +39,467 @@ example to construct such a store for the local filesystem.
+    structuring of the ``kartothek`` modules. In :mod:`kartothek.io` we have all
+    the building blocks to build data pipelines that read and write from/to storages.
+    The next module level (e.g. ``eager``) describes the scheduling backend.
+
+    The scheduling backends supported by kartothek are:
+
+    - ``eager`` runs all execution immediately and on the local machine.
+    - ``iter`` executes operations on the dataset on a per-partition basis.
+      The standard format to read/store dataframes in ``iter`` is by providing
+      a generator of dataframes.
+    - ``dask`` is suitable for larger datasets. It can be used to work on datasets in
+      parallel or even in a cluster by using ``dask.distributed`` as the backend.
+      There are also ``dask.bag`` and ``dask.dataframe`` which support I/O operations
+      for the respective `dask`_ objects.
 
 After calling :func:`~kartothek.io.eager.store_dataframes_as_dataset`,
-a :class:`kartothek.core.dataset.DatasetMetadata` object is returned. This is the main
-class holding all information about the parts and schema of the dataset.
+a :class:`kartothek.core.dataset.DatasetMetadata` object is returned. 
+This class holds information about the structure and schema of the dataset.
 
-The most interesting ones for this guide are ``tables`` and ``partitions``.
-Each dataset can have multiple tables, each table is a collection of files
-that all have the same schema. These files are called ``partitions`` in
-``kartothek``.
+For this guide, two attributes that are noteworthy are ``tables`` and ``partitions``:
 
-As we neither have explicitly defined the name of the table nor the name
+- Each dataset has one or more ``tables``, where each table represents a particular subset of
+  data, this data is stored as a collection of dataframes/files which have the same schema.
+- Data is written to storage in batches (for ``eager``, there is only a single batch),
+  in this sense a batch is termed a ``partition`` in ``kartothek``. An alternative to
+  this approach of partitioning takes place when the user specifies the column on which
+  they would like the dataset to be partitioned. In this case, there will exist a partition
+  for each different value of that column, for each batch of data.
+  Partitions are structurally identical to each other and each partition
+  is made up of a collection of datframes/files containing the subset of data of each table
+  belonging to that partition.
+
+
+For each table, ``kartothek`` also tracks the schema of the columns.
+Unless specified explicitly on write, it is inferred from the passed data.
+On writing additional data to a dataset, we will also check that the schema
+of the new data matches the schema of the existing data.
+
+.. admonition:: Passing multiple partitions to a dataset during write/update
+
+    To store multiple dataframes into a dataset (i.e. multiple `partitions`), it is possible
```

Comment:
This is also ambiguous. Storing multiple dataframes into a dataset could be misunderstood as writing _different_ dataframes into a table. You should be explicit that _multiple partitions_ means _multiple data chunks_ with the same schema

Reply:
Personally I am okay with the ambiguity exactly because it does encapsulate a bunch of possibilities. Thus the reason for this being within an admonition. The bigger idea being to convey that 'of course you can pass in multiple dataframes in one call to a write function but if and when you do so please be aware of how `kartothek` behaves in such cases so that you are not caught off-guard'.